l1-contracts/deploy-scripts/LensScript.s.sol

```diff
@@ -0,0 +1,292 @@
+            IGovernance.Call({
+                target: config.tempValidatorTimelock,
+                value: 0,
+                data: abi.encodeCall(Ownable2Step.acceptOwnership, ())
+            })
+        );
+    }
+
+    function _generateRegisterChainCall() internal {
+        calls.push(
+            IGovernance.Call({
+                target: config.newStateTransitionManager,
+                value: 0,
+                data: abi.encodeCall(
+                    IStateTransitionManager.registerAlreadyDeployedHyperchain,
+                    (config.lensChainId, config.lensDiamondProxy)
+                )
+            })
+        );
+    }
+
+    function _generateCreateChainCall() internal {
+        calls.push(
+            IGovernance.Call({
+                target: config.newBridgehub,
+                value: 0,
+                data: abi.encodeCall(
+                    IBridgehub.createNewChain,
+                    (
+                        config.lensChainId,
+                        config.lensDiamondProxy,
+                        config.baseToken,
+                        uint256(0),
+                        address(0),
+                        "0x"
+                    )
+                )
+            })
+        );
+    }
+
+    function _generateExecuteUpgradeCall() internal {
+        Diamond.FacetCut[] memory facetCuts = new Diamond.FacetCut[](0);
+
+        MigrationParams memory migrationParams = MigrationParams({
+            newVerifier: config.newVerifier,
+            newCTM: config.newStateTransitionManager,
+            newBridgehub: config.newBridgehub
+        });
+
+        Diamond.DiamondCutData memory diamondCut = Diamond.DiamondCutData({
+            facetCuts: facetCuts,
+            initAddress: migrationUpgradeAddress,
+            initCalldata: abi.encode(migrationParams)
+        });
+
+        calls.push(
+            IGovernance.Call({
+                target: config.lensDiamondProxy,
+                value: 0,
+                data: abi.encodeCall(IAdmin.executeUpgrade, (diamondCut))
+            })
+        );
+    }
+
+    function _printTempGovernanceScheduleOperation() internal view {
+        console.log("Temp Governance Schedule Operation:");
+        console.logBytes(tempEcoScheduleOperation);
+        console.log();
+    }
+
+    function _printTempGovernanceExecuteOperation() internal view {
+        console.log("Temp Governance Execute Operation:");
+        console.logBytes(tempEcoExecuteOperation);
+        console.log();
+    }
+
+    function _printMigrationExecuteUpgradeCall() internal view {
+        console.log("Migration Execute Upgrade Call:");
+
+        for (uint256 i = 0; i < calls.length; ++i) {
+            console.log(calls[i].target);
+            console.log(calls[i].value);
+            console.logBytes(calls[i].data);
+            console.log();
+        }
+    }
+}
+
+// todo
+// do we need to add a call to add the new validator to the validator timelock?
```

Comment:
it makes sense to generate the calldata just to make Len's life a bit easier, agree, but it should be a not part of the governance calldata